src/main/java/org/gnieh/logback/config/ConfigLoader.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * This file is part of the logback-config project.
+ * Copyright (c) 2018 Lucas Satabin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gnieh.logback.config;
+
+import com.typesafe.config.Config;
+
+/**
+ * Typesafe config loader interface.
+ */
+public interface ConfigLoader {
```

Comment:
It can be interesting to annotate that interface as functional, to give a hint on the intended usage.

```suggestion
@FunctionalInterface
public interface ConfigLoader {
```